cleanlab/segmentation/rank.py

```diff
@@ -0,0 +1,258 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Methods to rank and score images in a semantic segmentation dataset, based on how likely they are to contain label errors.
+
+"""
+import numpy as np
+from typing import Optional, Union, Tuple
+from cleanlab.segmentation.filter import find_label_issues
+from cleanlab.segmentation.filter import _check_input
+
+
+def get_label_quality_scores(
+    labels: np.ndarray,
+    pred_probs: np.ndarray,
+    *,
+    method: str = "softmin",
+    batch_size: int = 10000,
+    n_jobs: int = 1,
+    verbose: bool = True,
+    **kwargs,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Returns a label quality score for each image.
+
+    This is a function to compute label quality scores for semantic segmentation datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    N - Number of images in the dataset
+    K - Number of classes in the dataset
+    H - Height of each image
+    W - Width of each image
+
+    Parameters
+    ----------
+    labels : np.ndarray
+      A discrete array of noisy labels for a segmantic segmentation dataset, in the shape``(N,H,W,)``.
+      where each pixel must be integer in 0, 1, ..., K-1.
+
+    pred_probs : np.ndarray
+      An array of shape ``(N,K,H,W,)`` of model-predicted class probabilities,
+
+    Refer to documentation for this argument in :py:func:find_label_issues <cleanlab.segemntation.filter.find_label_issues>
+
+    method : {"softmin", "num_pixel_issues"}, default="softmin"
+      Label quality scoring method.
+        "softmin" - Calculates the inner product between scores and softmax(1-scores). For efficiency, use instead of "num_pixel_issues".
+        "num_pixel_issues" - Uses the number of pixels with label issues for each image using :py:func:find_label_issues <cleanlab.segemntation.filter.find_label_issues>
+
+    batch_size : int,
+      For num_pixel_issues:
+      Size of mini-batches to use for estimating the label issues.
+      To maximize efficiency, try to use the largest `batch_size` your memory allows.
+
+    n_jobs: int,
+      For num_pixel_issues:
+      Number of processes for multiprocessing (default value = 1). Only used on Linux.
+      If `n_jobs=None`, will use either the number of: physical cores if psutil is installed, or logical cores otherwise.
+
+    verbose : bool,
+      For num_pixel_issues:
+      Set to ``False`` to suppress all print statements.
+
+    **kwargs:
+      downsample : int,
+      Factor to shrink labels and pred_probs by for 'num_pixel_issues' only . Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear
+      decrease in performance
+
```

Comment:
what about the batch_size for find_label_issues()

Reply:
Sorry I don't follow - find_label_issues has a batch_size argument

Reply:
In that I am saying that the downsample does not work for the softmin method - maybe I should clarify that

Reply:
please don't resolve conversations involving a question, I will not see answer if you resolve it.

Here you are allowing user to specify `downsample` argument via `get_label_quality_scores()`, and I am proposing you should also allow user to specify `batch_size` argument. I get that both arguments would only apply when the method is  num_pixel_issues and would be ignored if it is softmin. 

And yes please explicitly mention that in docstring

Reply:
right - there is a batch_size argument for get_label_quality_scores() - am I misunderstanding something?

Reply:
sorry! resolved cause both get_label_quality_scores() and find_label_issues() both have a downsamples and batch_size argument

Reply:
Do you mean I should offer downsample in get_label_quality_scores() as an argument and not as a kwarg?

Reply:
They should both be kwargs or not. It does not make sense for one to be kwarg and the other not. I prefer both being kwargs. 
 
They should both be at the end of the arguments list too and docstring should clarify:  These arguments are only considered when method = "num_pixel_issues"

Reply:
generally don't split up arguments that only apply in a special case in the arg order between other args that always apply

Reply:
I think we talked about this earlier and this is what we came to because for find_label_issues() needs downsampling to be worthwhile and the default method for get_label_quality_score() is softmin.

If we make them both kwargs then what would you suggest the default downsample value to be? Right now it is 16, but if that number is not divisible by their image dimensions then it will fail and the user would have to go then set the kwarg which seems annoying to do. 

If we make downsample a kwarg for both methods, then I feel the default should be 1 or 2. Which are more likely to be a factor of the dimension. Thoughts?